### PR TITLE
[sw/vendor] Vendor-in cryptoc library

### PR DESCRIPTION
**Pull request by moidx**
Closes issue #999.

* Remove manually copied sw/vendor/cryptoc folder.
* Add cryptoc vendor manifest.
*  Update code from upstream repository
    https://chromium.googlesource.com/chromiumos/third_party/cryptoc/ to
    revision e05bfa91102dd5137b4027b4f3405e041ffe2c32
* Update include paths in header files after vendor in sw/vendor/cryptoc changes.

### CODE REVIEW

File: sw/vendor/cryptoc/OWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+mnissler@chromium.org
```
Comment (mcy) [inline sw/vendor/cryptoc/OWNERS]: I'm a little amused that we're vendoring in an OWNERS file. =P

Comment (moidx) [reply sw/vendor/cryptoc/OWNERS]: Should I remove the OWNERS file?

Comment (gkelly) [reply sw/vendor/cryptoc/OWNERS]: No, not worth adding patches just to do a cosmetic change.

**Comment by cdgori** [comment]
Is this code to execute on the Ibex?  I reviewed semi-carefully, there are some functions that have self-documented issues with point-at-infinity and a few of the other discontinuous points (comes up a lot in p256 and similar curves, to my limited recollection).  

Basically suggesting we have an EC-expert read through this and make sure it will be OK for what we are intending to use it for, as there are a lot of ways for things to go badly with carefully-crafted inputs using these weird points.  Error handling for EC is always hard to get correct because of these weird points.

**Comment by moidx** [comment]
> Is this code to execute on the Ibex? I reviewed semi-carefully, there are some functions that have self-documented issues with point-at-infinity and a few of the other discontinuous points (comes up a lot in p256 and similar curves, to my limited recollection).
> 
> Basically suggesting we have an EC-expert read through this and make sure it will be OK for what we are intending to use it for, as there are a lot of ways for things to go badly with carefully-crafted inputs using these weird points. Error handling for EC is always hard to get correct because of these weird points.

Hi @cdgori, At the moment we are only using interface definitions for SHA256.  This change is configuring the vendor script to pull the code from the original chromium repo. EC largely uses this library for interface definitions as well.  Even though this library can run on Ibex, I don't foresee being used in production. I will open an issue to keep track of this discussion. I think in the long run we can develop our own interfaces, and remove this dependency if it makes sense.

**Comment by cdgori** [comment]
@moidx SGTM, the SHA256 is much less controversial :)

File: sw/vendor/cryptoc/OWNERS (hunk 1 above)
Comment (tjaychen) [reply sw/vendor/cryptoc/OWNERS]: isn't there a way to exclude specific files during the vendor process?

**Pull request merged**